arm9/source/filesys/fsperm.c

```diff
@@ -26,6 +26,12 @@ bool CheckWritePermissions(const char* path) {
     if ((drvtype & DRV_IMAGE) && !CheckWritePermissions(GetMountPath()))
         return false; // endless loop when mounted file inside image, but not possible
     
+    // SD card write lock check
+    if ((drvtype & (DRV_IMAGE | DRV_SDCARD | DRV_EMUNAND | DRV_ALIAS)) && SD_WRITE_LOCKED) {
```

Comment:
Images can already have their origin on CTRNAND. You need to check the drive type of the image source. Besides, you don't need to check for images here. See the instrcuction block right above?